cargo-dist/src/init.rs

```diff
@@ -1449,43 +1386,457 @@ fn apply_dist_to_metadata(metadata: &mut toml_edit::Item, meta: &DistMetadata) {
+        common.install_libraries.as_ref(),
     );
 
+    // / Aliases to install binaries as
+    // FIXME/TODO: pub bin_aliases: Option<SortedMap<String, Vec<String>>>,
+}
+
+fn apply_installers_homebrew(installers_table: &mut toml_edit::Table, homebrew: &HomebrewInstallerLayer) {
+    let Some(homebrew_table) = installers_table.get_mut("homebrew") else {
+        return;
+    };
+    let toml_edit::Item::Table(homebrew_table) = homebrew_table else {
+        panic!("Expected [dist.installers.homebrew] to be a table");
+    };
+
+    apply_installers_common(homebrew_table, &homebrew.common);
+
+    // TODO: similar to shell
+}
+
+fn apply_installers_msi(installers_table: &mut toml_edit::Table, msi: &MsiInstallerLayer) {
+    let Some(msi_table) = installers_table.get_mut("msi") else {
+        return;
+    };
+    let toml_edit::Item::Table(msi_table) = msi_table else {
+        panic!("Expected [dist.installers.msi] to be a table");
+    };
+
+    apply_installers_common(msi_table, &msi.common);
+
+    // TODO: similar to shell
+}
+
+fn apply_installers_npm(installers_table: &mut toml_edit::Table, npm: &NpmInstallerLayer) {
+    let Some(npm_table) = installers_table.get_mut("npm") else {
+        return;
+    };
+    let toml_edit::Item::Table(npm_table) = npm_table else {
+        panic!("Expected [dist.installers.npm] to be a table");
+    };
+
+    apply_installers_common(npm_table, &npm.common);
+
+    // TODO: similar to shell
+}
+
+fn apply_installers_powershell(installers_table: &mut toml_edit::Table, powershell: &PowershellInstallerLayer) {
+    let Some(powershell_table) = installers_table.get_mut("powershell") else {
+        return;
+    };
+    let toml_edit::Item::Table(powershell_table) = powershell_table else {
+        panic!("Expected [dist.installers.powershell] to be a table");
+    };
+
+    apply_installers_common(powershell_table, &powershell.common);
+
+    // TODO: similar to shell
+}
+
+fn apply_installers_shell(installers_table: &mut toml_edit::Table, shell: &ShellInstallerLayer) {
+    let Some(shell_table) = installers_table.get_mut("shell") else {
+        return;
+    };
+    let toml_edit::Item::Table(shell_table) = shell_table else {
+        panic!("Expected [dist.installers.shell] to be a table");
+    };
+
+    apply_installers_common(shell_table, &shell.common);
+
+    // TODO
+}
+
+fn apply_installers_pkg(installers_table: &mut toml_edit::Table, pkg: &PkgInstallerLayer) {
+    let Some(pkg_table) = installers_table.get_mut("pkg") else {
+        return;
+    };
+    let toml_edit::Item::Table(pkg_table) = pkg_table else {
+        panic!("Expected [dist.installers.pkg] to be a table");
+    };
+
+    apply_installers_common(pkg_table, &pkg.common);
+
+    apply_optional_value(
+        pkg_table,
+        "identifier",
+        "# A unique identifier, in tld.domain.package format\n",
+        pkg.identifier.clone(),
+    );
+
+    apply_optional_value(
+        pkg_table,
+        "install_location",
```

Comment:
Typo: this one is still kebab-case.
```suggestion
        "install-location",
```